### ISSUE TITLE
Unable to parse elastic response with pretty=true

### ISSUE THREAD

**Issue by topres** [elastic/elasticsearch-net#2294]
**NEST/Elasticsearch.Net version**: 5.0.0-beta1

**Elasticsearch version**: 5.0.0-beta1

The Elasticsearch.Net low level client does not work when pretty=true is enabled on my clients connection settings. When I disable pretty=true, it works just fine. 

Below here I have linked a pastebin containing a json document which was returned from ElasticSearch with pretty=true and could not be parsed by my nest client.

http://pastebin.com/L7wzeJ1i

The following response is returned in DebugInformation:

"Unsuccessful low level call on GET: /customer_index/product/_search?pretty=true\r\n# Audit trail of this API call:\r\n - [1] AllNodesDead: Took: 00:00:00\r\n - [2] Resurrection: Node: http://192.168.0.39:9200/ Took: 00:00:00\r\n - [3] PingSuccess: Node: http://192.168.0.39:9200/ Took: 00:00:00.2932808\r\n - [4] BadResponse: Node: http://192.168.0.39:9200/ Took: 00:00:00.0005026\r\n# OriginalException: System.Net.WebException: The server cause a ProtocolViolationException. Section=ResponseStatusLine\r\n   ved System.Net.HttpWebRequest.GetResponse()\r\n   ved Elasticsearch.Net.HttpConnection.Request[TReturn](RequestData requestData)\r\n# Request:\r\n<Request stream not captured or already read to completion by serializer. Set DisableDirectStreaming() on ConnectionSettings to force it to be set on the response.>\r\n# Response:\r\n<Response stream not captured or already read to completion by serializer. Set DisableDirectStreaming() on ConnectionSettings to force it to be set on the response.>\r\n"

**Comment by gmarz** [comment]
@topres The thing that jumps out right away is the `GET` request, rather than `POST`.  .NET doesn't allow a body for `GET` requests.

Can you paste the relevant client code here?  Mainly the search request and your connection settings.

**Comment by niemyjski** [comment]
I reproduce this on beta1 as well when I added `PrettyJson` to connection settings

```csharp
settings.EnableTcpKeepAlive(TimeSpan.FromSeconds(30), TimeSpan.FromSeconds(2)).DisableDirectStreaming().PrettyJson();
```

```
System.ApplicationException
Error checking to see if index identity exists: Original: ( - WebException) The server committed a protocol violation. Section=ResponseStatusLine

   at Foundatio.Repositories.Elasticsearch.Configuration.IndexBase.<IndexExistsAsync>d__20.MoveNext() in C:\Code\Foundatio.Repositories\src\Foundatio.Repositories.Elasticsearch\Configuration\IndexBase.cs:line 136
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Foundatio.Repositories.Elasticsearch.Configuration.IndexBase.<DeleteIndexAsync>d__19.MoveNext() in C:\Code\Foundatio.Repositories\src\Foundatio.Repositories.Elasticsearch\Configuration\IndexBase.cs:line 108
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   at Foundatio.Repositories.Elasticsearch.Configuration.ElasticConfiguration.<DeleteIndexesAsync>d__32.MoveNext() in C:\Code\Foundatio.Repositories\src\Foundatio.Repositories.Elasticsearch\Configuration\IElasticConfiguration.cs:line 158
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at Foundatio.Repositories.Elasticsearch.Tests.ElasticRepositoryTestBase.<RemoveDataAsync>d__6.MoveNext() in C:\Code\Foundatio.Repositories\tests\Foundatio.Repositories.Elasticsearch.Tests\ElasticRepositoryTestBase.cs:line 40
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at Foundatio.Repositories.Elasticsearch.Tests.ReadOnlyRepositoryTests..ctor(ITestOutputHelper output) in C:\Code\Foundatio.Repositories\tests\Foundatio.Repositories.Elasticsearch.Tests\ReadOnlyRepositoryTests.cs:line 25
```

Here is the calling code:
```cs
        protected async Task<bool> IndexExistsAsync(string name) {
            if (name == null)
                throw new ArgumentNullException(nameof(name));

            var response = await Configuration.Client.IndexExistsAsync(name).AnyContext();
            if (response.IsValid)
                return response.Exists;

            string message = $"Error checking to see if index {name} exists: {response.GetErrorMessage()}";
            _logger.Error().Exception(response.OriginalException).Message(message).Property("request", response.GetRequest()).Write();
            throw new ApplicationException(message, response.OriginalException);
        }
```

**Comment by niemyjski** [comment]
```
Invalid NEST response built from a unsuccessful low level call on HEAD: /identity?pretty=true
# Audit trail of this API call:
 - [1] PingSuccess: Node: http://ipv4.fiddler:9200/ Took: 00:00:00.4210542
 - [2] BadResponse: Node: http://ipv4.fiddler:9200/ Took: 00:00:00.1050037
# OriginalException: System.Net.WebException: The server committed a protocol violation. Section=ResponseStatusLine
   at System.Net.HttpWebRequest.EndGetResponse(IAsyncResult asyncResult)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncCoreLogic(IAsyncResult iar, Func`2 endFunction, Action`1 endAction, Task`1 promise, Boolean requiresSynchronization)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Elasticsearch.Net.HttpConnection.<RequestAsync>d__7`1.MoveNext() in C:\code\elasticsearch-net\src\Elasticsearch.Net\Connection\HttpConnection.cs:line 179
# Request:
<Request stream not captured or already read to completion by serializer. Set DisableDirectStreaming() on ConnectionSettings to force it to be set on the response.>
# Response:
<Response stream not captured or already read to completion by serializer. Set DisableDirectStreaming() on ConnectionSettings to force it to be set on the response.>
```

**Comment by topres** [comment]
@gmarz The issue occurs both with GET and POST requests. If you need more info, I'll be happy to see if I can find some time to reproduce it and send you more data. I tried the following:

GET a single document with NEST.
GET all documents in a index/type with the LowLevel client.
POST search all documents in a index/type with the high level NEST api.

All failed with the same error, until I removed pretty=true.

**Comment by Mpdreamz** [comment]
@topres @niemyjski are you running elasticsearch server on linux? I suspect it might be CRLF header parsing issue.

If you can [enable network tracing](https://msdn.microsoft.com/en-us/library/a6sbz1dx(v=vs.110).aspx) or have a wireshark dump of when it happens that be really helpful. 

So far not been successful reproducing the issue.

**Comment by niemyjski** [comment]
These errors are all happening for me on local windows instances

**Comment by niemyjski** [comment]
If you want a running repo case, you could try our repo (I know that sucks to say and I'd love to have a snippet for you but literally every one of my tests breaks)

**Comment by gmarz** [comment]
Turns out this is an issue with Elasticsearch returning a response body on `HEAD` requests with `pretty=true`.  If you're using connection pooling with pinging enabled (default) and pretty json enabled, you'll hit this as .NET conforms strictly to the HTTP 1.1 spec.

I opened https://github.com/elastic/elasticsearch/issues/21075.

**Comment by gmarz** [comment]
This will be fixed in the next release of ES 5.0.

Closing.  Thanks for reporting @topres and @niemyjski.
